cmake/MachineCfg.cmake

```diff
@@ -40,6 +49,13 @@ if (USE_MACHINE_CFG)
 endif()
 
 if (MACHINE_CFG_FULL)
+  if (CMAKE_PROJECT_NAME)
+    message(WARNING "MachineCfg.cmake should be included before calling the \
+      project() function. If you're using parthenon as a subdirectory, try \
+      adding an include for it in your top level CMakeLists.txt prior to \
+      calling project().")
+  endif()
```

Comment:
Are you referring to this? https://cmake.org/cmake/help/latest/variable/CMAKE_PROJECT_PROJECT-NAME_INCLUDE_BEFORE.html#variable:CMAKE_PROJECT_%3CPROJECT-NAME%3E_INCLUDE_BEFORE

Reply:
I don't quite understand. When would this code be hit? Wouldn't the function have returned already?

Reply:
No, it's only returned if the machine config file is included more than once. This warning gets triggered if the project name is defined before you include the machine config file.

Reply:
The `if (CMAKE_PROJECT_NAME)` part I understand. But this is encased in an `if (MACHINE_CFG_FULL)` clause too. Won't that always return false? Or am I missing something?

Reply:
No because if its the first time through the file MACHINE_CFG_FULL is only defined after the return logic.

Reply:
right so if it's undefined, the if clause will read it as false, and this won't happen. But if it's defined as true, then the list would have already exited out, right?

Reply:
But it gets defined on line 40 before the if clause on line 51 but after the return statement on line 15.

Reply:
Oh I see. I missed that. Sorry for the confusion.

Reply:
Ya, no problem its easy to miss.

Reply:
> Are you referring to this? https://cmake.org/cmake/help/latest/variable/CMAKE_PROJECT_PROJECT-NAME_INCLUDE_BEFORE.html#variable:CMAKE_PROJECT_%3CPROJECT-NAME%3E_INCLUDE_BEFORE

Hm... this seems more like an alternative to `MACHINE_CFG`. `MACHINE_CFG` really needs to be included before *any* project command is called (or, at least, any language is enabled), not just before the `project(parthenon)` call.